File: quartz-energy/src/backend/QuartzEnergy.FracSchedule.Services/Vega/Models/Vehicles/Overviews/VehicleOverviewsRequest.cs
namespace QuartzEnergy.FracSchedule.Services.Vega.Models.Vehicles.Overviews
{
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;

    using QuartzEnergy.Common.DataAnnotations.Attributes.Data;
    using QuartzEnergy.Common.Services.Models.Overviews;

    public sealed class VehicleOverviewsRequest : OverviewsRequestModel
    {
        public VehicleOverviewsRequest(
            IEnumerable<int> makers,
            IEnumerable<int> models,
            IEnumerable<int> features,
            string contactName,
            string sortBy, 
            bool desc,
            int? pageSize, 
            int? pageNumber)
            : base(sortBy, desc, pageSize, pageNumber)
        {
            this.Makers = makers;
            this.Models = models;
            this.Features = features;
            this.ContactName = contactName;
        }

        [IntIds]
        public IEnumerable<int> Makers { get; }

        [IntIds]
        public IEnumerable<int> Models { get; }

        [IntIds]
        public IEnumerable<int> Features { get; }

        [MaxLength(40)]
        public string ContactName { get; }
    }
}

File: quartz-energy/src/frontend/src/app/vega/models/vehicles/overviews/vehicle-overviews-request.ts
import { OverviewsRequest } from '../../../../common/models/overviews/overviews-request';

export class VehicleOverviewsRequest extends OverviewsRequest {
    constructor (
        public makers: number[],
        public models: number[],
        public features: number[],
        public contactName: string,
        public sortBy: string,
        public desc: boolean,
        public pageSize: number,
        public pageNumber: number) {
      super(sortBy, desc, pageSize, pageNumber);
    }
}
